documentation/swagger-checklist.md

```diff
@@ -43,82 +45,236 @@ This document defines a Swagger checklist for Microsoft Azure, along with clarif
 
 Normative statements of the checklist are presented in the following manner:
 
-&#x25a2; *Mnnnn*: checklist text here.
+- [ ] **Mnnnn**: checklist text here.
 
 OR
 
-&#x25a2; *Snnnn*: checklist text here.
+- [ ] **Snnnn**: checklist text here.
 
-Where "M" stands for 'MUST' or 'MUST NOT' requirement. "S" stands for 'SHOULD' or 'SHOULD NOT' requirement. And, "nnnn" is replaced by a number that is unique among all the requirements in the checklist, thereby "nnnn" becomes a unique requirement identifier.
+Where "M" stands for '*MUST*' or '*MUST NOT*' requirement. "S" stands for '*SHOULD*' or '*SHOULD NOT*' requirement. And, "nnnn" is replaced by a number that is unique among all the requirements in the checklist, thereby "nnnn" becomes a unique requirement identifier.
 
 
 ## Naming - Swagger Checklist ##
 
-&#x25a2; *M1000*: Plural names MUST only be used to represent collections.
+- [ ] **M1000**: Plural names *MUST* only be used to represent collections.
+
+- [ ] **M1001**: Each operation must contain an OperationId of the form "RESOURCE_METHOD" a.k.a "NOUN_VERB", where "RESOURCE" represents the resource or object the operation addresses, and METHOD is a short name for the operation being performed "Get, Start, Delete, Create, List", etc.
+
+- [ ] **M1002**: If the NOUN in the OperationId "NOUN_VERB" matches with the name of the Model definition then the NOUN in the operationId *MUST* use the plural form of NOUN to avoid collision in the namespace.
+
+- [ ] **M1003**: List operations *MUST* be prefixed with "List" and *MUST NOT* be named as "GetXXXX" operations.
+
+- [ ] **M1004**: List operations *MUST* use consistent names and semantics. List operations *MUST* NOT use any other names. Consistent names and semantics are:
+
+  - `List()` - lists all resources under a subscription.
+  - `ListByResourceGroup()` - list all resources in a resource group within a subscription.
+  - `ListByParent()` - where "Parent" is a context specific suffix. It lists all resource under a parent.
+
+- [ ] **M1005**: Get operations *MUST* use the METHOD name "Get"
+
+- [ ] **M1006**: PUT operations *MUST* use the METHOD Name "Create"
+
+- [ ] **M1007**: PATCH operations *MUST* use the METHOD Name "Update"
+
+- [ ] **M1008**: POST operations *MUST* use a METHOD Name that matches the semantic operation on the resource, for example "Start" or "Stop"
 
-&#x25a2; *M1001*: List operations MUST be prefixed with "List" and MUST not be named as "GetXXXX" operations.
+- [ ] **M1009**: DELETE operations *MUST* use the METHOD Name "Delete"
 
-&#x25a2; *M1002*: List operations MUST use consistent names and semantics. List operations MUST NOT use any other names. Consistent names and semantics are:
+- [ ] **M1010**: Terminology in Swagger and Azure Portal *MUST* be the same.
```

Comment:
Then it cannot be a must. Unless you can also make it a must that names don't change after an API has shipped. Or you are willing to take massive breaking changes/renames between versions if a name changes. We've been through this cycle a number of times, and there is always a long discussion to be had before we choose to align terminology between SDKs and portal lingo.